# Project 3 Part 2 - Business Problem
- Exploratory Data Analysis
- Steven Phillips
- January 9, 2023
- Part 2: Use an API to extract box office revenue and profit data to add to your IMDB data and perform exploratory data analysis.

## Exploratory Data Analysis

In [1]:
import pandas as pd

### Load in your csv.gz's of results for each year extracted.

In [2]:
# Open 2000 saved file and preview again
df_tmdb_2000 = pd.read_csv("Data/final_tmdb_data_2000.csv.gz", low_memory = False)
df_tmdb_2000.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0116748    0.0  /wr0hTHwkYIRC82MwNbhOvqrw27N.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]      NaN   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN   
4         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

         id original_language   original_title  ... revenue  runtime  \
0       NaN               NaN              NaN  ...     NaN      NaN   
1   62127.0                en  The Fantasticks  ...     0.0     86.0   
2  110977.0                en    For the Cause  ...     0.0    100.0   
3  442869.0                hi             Gang  ...     0.0    152.0   
4  579396.0                hi         Karobaar  ...     0.0    180.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   

                                         tagline            title  video  \
0                                            NaN              NaN    NaN   
1  Try to remember the first time magic happened  The Fantasticks    0.0   
2   The ultimate showdown on a forbidden planet.    For the Cause    0.0   
3                                            NaN             Gang    0.0   
4                           The Business of Love         Karobaar    0.0   

   vote_average vote_count certification  
0           NaN        NaN           NaN  
1           5.5       22.0           NaN  
2           5.1        8.0           NaN  
3           4.0        1.0           NaN  
4           5.5        2.0           NaN  

[5 rows x 26 columns]

In [3]:
# Open 2001 saved file and preview again
df_tmdb_2001 = pd.read_csv("Data/final_tmdb_data_2001.csv.gz", low_memory = False)
df_tmdb_2001.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0035423    0.0  /hfeiSfWYujh6MKhtGTXyK3DD4nN.jpg                   NaN   
2  tt0079644    0.0  /79axmuH1UGkB7m72jjB9rPff9om.jpg                   NaN   
3  tt0089067    0.0                               NaN                   NaN   
4  tt0114447    0.0                               NaN                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  48000000.0  [{'id': 10749, 'name': 'Romance'}, {'id': 14, ...      NaN   
2         0.0                     [{'id': 10752, 'name': 'War'}]      NaN   
3         0.0                     [{'id': 35, 'name': 'Comedy'}]      NaN   
4         0.0  [{'id': 53, 'name': 'Thriller'}, {'id': 28, 'n...      NaN   

         id original_language             original_title  ...     revenue  \
0       NaN               NaN                        NaN  ...         NaN   
1   11232.0                en             Kate & Leopold  ...  76019048.0   
2  285529.0                id              November 1828  ...         0.0   
3  210258.0                es  El día de los albañiles 2  ...         0.0   
4  151007.0                en           The Silent Force  ...         0.0   

   runtime                                   spoken_languages    status  \
0      NaN                                                NaN       NaN   
1    118.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2    140.0  [{'english_name': 'Indonesian', 'iso_639_1': '...  Released   
3     90.0  [{'english_name': 'Spanish', 'iso_639_1': 'es'...  Released   
4     90.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline  \
0                                                NaN   
1  If they lived in the same century, they'd be p...   
2                                                NaN   
3     The laborers are back full of love and laughs.   
4  They left him for dead... They should have fin...   

                       title  video  vote_average vote_count certification  
0                        NaN    NaN           NaN        NaN           NaN  
1             Kate & Leopold    0.0          6.32     1164.0         PG-13  
2              November 1828    0.0          0.00        0.0           NaN  
3  El día de los albañiles 2    0.0          7.20       71.0           NaN  
4           The Silent Force    0.0          5.00        3.0           NaN  

[5 rows x 26 columns]

### Concatenate the data into 1 data frame for the remainder of the analysis.

In [4]:
df_tmdb_combined = pd.concat([df_tmdb_2000, df_tmdb_2001])
df_tmdb_combined.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0116748    0.0  /wr0hTHwkYIRC82MwNbhOvqrw27N.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]      NaN   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN   
4         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

         id original_language   original_title  ... revenue  runtime  \
0       NaN               NaN              NaN  ...     NaN      NaN   
1   62127.0                en  The Fantasticks  ...     0.0     86.0   
2  110977.0                en    For the Cause  ...     0.0    100.0   
3  442869.0                hi             Gang  ...     0.0    152.0   
4  579396.0                hi         Karobaar  ...     0.0    180.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   

                                         tagline            title  video  \
0                                            NaN              NaN    NaN   
1  Try to remember the first time magic happened  The Fantasticks    0.0   
2   The ultimate showdown on a forbidden planet.    For the Cause    0.0   
3                                            NaN             Gang    0.0   
4                           The Business of Love         Karobaar    0.0   

   vote_average vote_count certification  
0           NaN        NaN           NaN  
1           5.5       22.0           NaN  
2           5.1        8.0           NaN  
3           4.0        1.0           NaN  
4           5.5        2.0           NaN  

[5 rows x 26 columns]

In [5]:
# shape of the concatenated data frame
df_tmdb_combined.shape

(4470, 26)

In [6]:
# make a working copy of the data frame
df_movies = df_tmdb_combined.copy()
df_movies.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0116748    0.0  /wr0hTHwkYIRC82MwNbhOvqrw27N.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]      NaN   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN   
4         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

         id original_language   original_title  ... revenue  runtime  \
0       NaN               NaN              NaN  ...     NaN      NaN   
1   62127.0                en  The Fantasticks  ...     0.0     86.0   
2  110977.0                en    For the Cause  ...     0.0    100.0   
3  442869.0                hi             Gang  ...     0.0    152.0   
4  579396.0                hi         Karobaar  ...     0.0    180.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   

                                         tagline            title  video  \
0                                            NaN              NaN    NaN   
1  Try to remember the first time magic happened  The Fantasticks    0.0   
2   The ultimate showdown on a forbidden planet.    For the Cause    0.0   
3                                            NaN             Gang    0.0   
4                           The Business of Love         Karobaar    0.0   

   vote_average vote_count certification  
0           NaN        NaN           NaN  
1           5.5       22.0           NaN  
2           5.1        8.0           NaN  
3           4.0        1.0           NaN  
4           5.5        2.0           NaN  

[5 rows x 26 columns]

### Exploratory Data Analysis:

#### 1. How many movies had at least some valid financial information (values > 0 for budget OR revenue)?
- Please exclude any movies with 0's for budget AND revenue from the remaining visualizations.

In [7]:
# filter movies with only positive budget OR positive revenue
budget_filter = df_movies['budget'] > 0
revenue_filter = df_movies['revenue'] > 0

df_movies = df_movies.loc[budget_filter | revenue_filter, :]
df_movies.head()

imdb_id  adult                     backdrop_path  \
1   tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg   
5   tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
7   tt0119273    0.0  /f5C03doOWiauu37bToKXtpgP5bS.jpg   
13  tt0120263    0.0  /ynXVuylP8upazjz8lrqb1PEMkdR.jpg   
14  tt0120467    0.0  /knok3mNReKqPTplnnqz7E4dd7mD.jpg   

                                belongs_to_collection      budget  \
1                                                 NaN  10000000.0   
5                                                 NaN    150000.0   
7   {'id': 141086, 'name': 'Heavy Metal Collection...  15000000.0   
13                                                NaN         0.0   
14                                                NaN    120000.0   

                                               genres homepage       id  \
1   [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN  62127.0   
5   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN    843.0   
7   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...      NaN  16225.0   
13  [{'id': 18, 'name': 'Drama'}, {'id': 35, 'name...      NaN  34070.0   
14  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...      NaN  19085.0   

   original_language              original_title  ...     revenue  runtime  \
1                 en             The Fantasticks  ...         0.0     86.0   
5                 cn                        花樣年華  ...  12854953.0     99.0   
7                 en            Heavy Metal 2000  ...         0.0     88.0   
13                sv  Sånger från andra våningen  ...     80334.0     98.0   
14                en                      Vulgar  ...     14904.0     87.0   

                                     spoken_languages    status  \
1   [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
5   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
7   [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
13  [{'english_name': 'Russian', 'iso_639_1': 'ru'...  Released   
14  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                              tagline  \
1       Try to remember the first time magic happened   
5   Feel the heat, keep the feeling burning, let t...   
7   not to survive the fight in the external world...   
13                                                NaN   
14   Everyone loves a clown... some more than others.   

                          title  video  vote_average vote_count certification  
1               The Fantasticks    0.0         5.500       22.0           NaN  
5          In the Mood for Love    0.0         8.111     2078.0            PG  
7              Heavy Metal 2000    0.0         6.078      173.0             R  
13  Songs from the Second Floor    0.0         7.217      226.0           NaN  
14                       Vulgar    0.0         5.400       42.0             R  

[5 rows x 26 columns]

In [8]:
# shape of the filtered data frame

df_movies.shape

(728, 26)

1. There are 728 movies remaining with at least some financial information. (values > 0 for budget OR revenue.)

In [9]:
# filter only movies with BOTH positive budget and revenue

budget_filter = df_movies['budget'] > 0
revenue_filter = df_movies['revenue'] > 0


df_movies = df_movies.loc[budget_filter & revenue_filter, :]
df_movies.head()

imdb_id  adult                     backdrop_path  \
5   tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
14  tt0120467    0.0  /knok3mNReKqPTplnnqz7E4dd7mD.jpg   
17  tt0120630    0.0  /t7aJ7hqsMLEoqBJk7yAXxcAVvcz.jpg   
20  tt0120753    0.0  /3vUkQpnwbsFEQlMh9EORpvKJfpo.jpg   
21  tt0120755    0.0  /24DZfupDlhXeTchmcOkoGRhP5Vg.jpg   

                                belongs_to_collection       budget  \
5                                                 NaN     150000.0   
14                                                NaN     120000.0   
17  {'id': 718551, 'name': 'Chicken Run Collection...   45000000.0   
20                                                NaN    8000000.0   
21  {'id': 87359, 'name': 'Mission: Impossible Col...  125000000.0   

                                               genres homepage       id  \
5   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN    843.0   
14  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...      NaN  19085.0   
17  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...      NaN   7443.0   
20  [{'id': 18, 'name': 'Drama'}, {'id': 53, 'name...      NaN    318.0   
21  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...      NaN    955.0   

   original_language            original_title  ...      revenue  runtime  \
5                 cn                      花樣年華  ...   12854953.0     99.0   
14                en                    Vulgar  ...      14904.0     87.0   
17                en               Chicken Run  ...  224834564.0     84.0   
20                en  The Million Dollar Hotel  ...     105983.0    122.0   
21                en    Mission: Impossible II  ...  546388105.0    123.0   

                                     spoken_languages    status  \
5   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
14  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
17  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
20  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
21  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                              tagline  \
5   Feel the heat, keep the feeling burning, let t...   
14   Everyone loves a clown... some more than others.   
17  This ain't no chick flick. It's poultry in mot...   
20                    Everyone has something to hide.   
21                       Expect the impossible again.   

                       title  video  vote_average vote_count certification  
5       In the Mood for Love    0.0         8.111     2078.0            PG  
14                    Vulgar    0.0         5.400       42.0             R  
17               Chicken Run    0.0         6.735     4223.0             G  
20  The Million Dollar Hotel    0.0         5.904      260.0             R  
21    Mission: Impossible II    0.0         6.100     5596.0         PG-13  

[5 rows x 26 columns]

In [10]:
# shape of filtered data frame
df_movies.shape

(373, 26)

#### 2. How many movies are there in each of the certification categories (G/PG/PG-13/R)?

In [11]:
df_movies['certification'].value_counts()

R        144
PG-13    117
PG        30
G         13
NR         6
Name: certification, dtype: int64

2. There are 144 R movies, 117 PG-13 movies, 30 PG movies, 13 G movies, and also 6 NR movies.

#### 3. What is the average revenue per certification category?

In [12]:
df_movies.groupby('certification')['revenue'].mean()

certification
G        1.332169e+08
NR       2.382332e+07
PG       1.304904e+08
PG-13    1.110740e+08
R        5.230737e+07
Name: revenue, dtype: float64

3. The average revenue per certification category is 13,321,690 dollars for G movies, 2,383,332 dollars for NR movies, 13,049,040 dollars for PG movies, 11,107,400 dollars for PG-13 movies, and 5,230,737 dollars for R movies.

#### 4. What is the average budget per certification category?

In [13]:
df_movies.groupby('certification')['budget'].mean()

certification
G        4.400000e+07
NR       1.478333e+07
PG       5.218333e+07
PG-13    4.680682e+07
R        2.695496e+07
Name: budget, dtype: float64

3. The average budget per certification category is 4,400,000 dollars for G movies, 1,478,333 dollars for NR movies, 5,218,333 dollars for PG movies, 4,680,682 dollars for PG-13 movies, and 2,695,496 dollars for R movies.

## Save a final merged .csv.gz of all of the tmdb api data
- The file name should be "tmdb_results_combined.csv.gz"

In [14]:
FOLDER = "Data/"

In [15]:
df_tmdb_combined.to_csv(f"{FOLDER}tmdb_results_combined.csv.gz", compression="gzip", index=False)

In [16]:
# Open saved merged TMDB API data file and preview as data frame

df_tmdb_merged = pd.read_csv("Data/tmdb_results_combined.csv.gz", low_memory = False)
df_tmdb_merged.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0116748    0.0  /wr0hTHwkYIRC82MwNbhOvqrw27N.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]      NaN   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN   
4         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

         id original_language   original_title  ... revenue  runtime  \
0       NaN               NaN              NaN  ...     NaN      NaN   
1   62127.0                en  The Fantasticks  ...     0.0     86.0   
2  110977.0                en    For the Cause  ...     0.0    100.0   
3  442869.0                hi             Gang  ...     0.0    152.0   
4  579396.0                hi         Karobaar  ...     0.0    180.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   

                                         tagline            title  video  \
0                                            NaN              NaN    NaN   
1  Try to remember the first time magic happened  The Fantasticks    0.0   
2   The ultimate showdown on a forbidden planet.    For the Cause    0.0   
3                                            NaN             Gang    0.0   
4                           The Business of Love         Karobaar    0.0   

   vote_average vote_count certification  
0           NaN        NaN           NaN  
1           5.5       22.0           NaN  
2           5.1        8.0           NaN  
3           4.0        1.0           NaN  
4           5.5        2.0           NaN  

[5 rows x 26 columns]

In [17]:
df_tmdb_merged.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4470 entries, 0 to 4469
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                4470 non-null   object 
 1   adult                  4468 non-null   float64
 2   backdrop_path          1973 non-null   object 
 3   belongs_to_collection  251 non-null    object 
 4   budget                 4468 non-null   float64
 5   genres                 4468 non-null   object 
 6   homepage               231 non-null    object 
 7   id                     4468 non-null   float64
 8   original_language      4468 non-null   object 
 9   original_title         4468 non-null   object 
 10  overview               4116 non-null   object 
 11  popularity             4468 non-null   float64
 12  poster_path            3808 non-null   object 
 13  production_companies   4468 non-null   object 
 14  production_countries   4468 non-null   object 
 15  rele